pyfmt/select.py

```diff
@@ -0,0 +1,77 @@
+import subprocess
+from dataclasses import dataclass
+from typing import Iterator, Tuple
+
+__all__ = ["select_staged", "select_modified", "select_head", "select_local", "select_all"]
+
+
+def select_staged(path: str) -> Iterator[str]:
+    return (file for code, file in _iter_changed(path) if code.index_has_changes())
+
+
+def select_modified(path: str) -> Iterator[str]:
+    return (file for code, file in _iter_changed(path) if code.has_changes() or code.is_untracked())
+
+
+def select_head(path: str) -> Iterator[str]:
+    return _iter_committed(path, "HEAD^1..HEAD")
+
+
+def select_local(path: str) -> Iterator[str]:
+    return _iter_committed(path, "@{upstream}..")
+
+
+def select_all(path: str) -> Iterator[str]:
+    return [path]
+
+
+@dataclass
```

Comment:
nice

Reply:
this means everything needs to be 3.7 and above to use pyfmt. If `https://pypi.org/project/dataclasses/` can work as a drop in, we'd get 3.6 compat. Can you look into this a bit? It would be nice if there was another means rather than using this decorator since it really hampers compatibility. 

I think it would be best if `setup.py` reflects the supported python versions when we affect compatibility like this since it's used by folks across gooee (on older code bases) and folks are using it outside of Gooee.

Reply:
ah true. I can use something else

Reply:
Yeah may be worth defining what our earliest version supported is and avoid features not supported by older versions